contracts/SlashIndicator.sol

```diff
@@ -161,6 +185,93 @@ contract SlashIndicator is ISlashIndicator,System,IParamSubscriber, IApplication
     emit indicatorCleaned();
   }
 
+  function submitFinalityViolationEvidence(FinalityEvidence calldata _evidence) external onlyInit onlyRelayer {
+    if (finalitySlashRewardRatio == 0) {
```

Comment:
We need to ensure the `FinalityEvidence` can not be reused, I mean a validator is been slashed because of evidenceA, then it can not be slashed again because of evidenceA again.

Reply:
A potential solution is to check the evidence  happened in recent blocks.

Reply:
Fixed